tests/unit_tests/scheduler/test_openpbs_driver.py

```diff
@@ -40,3 +60,66 @@ async def test_no_validation_of_memory_per_job():
     driver = OpenPBSDriver(memory_per_job="a_lot")
     await driver.submit(0, "sleep")
     assert " -l mem=a_lot " in Path("captured_qsub_args").read_text(encoding="utf-8")
+
+
+@pytest.mark.usefixtures("capturing_qsub")
+async def test_num_nodes():
+    driver = OpenPBSDriver(num_nodes=2)
+    await driver.submit(0, "sleep")
+    resources = parse_resource_string(
+        Path("captured_qsub_args").read_text(encoding="utf-8")
+    )
+    assert resources["nodes"] == "2"
+
+
+@pytest.mark.usefixtures("capturing_qsub")
+async def test_num_nodes_default():
+    # This is the driver default, Ert can fancy a different default
+    driver = OpenPBSDriver()
+    await driver.submit(0, "sleep")
+    resources = parse_resource_string(
+        Path("captured_qsub_args").read_text(encoding="utf-8")
+    )
+    assert "nodes" not in resources
+
+
+@pytest.mark.usefixtures("capturing_qsub")
+async def test_num_cpus_per_node():
+    driver = OpenPBSDriver(num_cpus_per_node=2)
+    await driver.submit(0, "sleep")
+    resources = parse_resource_string(
+        Path("captured_qsub_args").read_text(encoding="utf-8")
+    )
+    assert resources["ppn"] == "2"
+
+
+@pytest.mark.usefixtures("capturing_qsub")
+async def test_num_cpus_per_node_default():
+    driver = OpenPBSDriver()
+    await driver.submit(0, "sleep")
+    resources = parse_resource_string(
+        Path("captured_qsub_args").read_text(encoding="utf-8")
+    )
+    assert "ppn" not in resources
+
+
+@given(words, st.integers(min_value=0), st.integers(min_value=0))
+@pytest.mark.usefixtures("capturing_qsub")
+async def test_full_resource_string(memory_per_job, num_nodes, num_cpus_per_node):
```

Comment:
Don't fully understand the meaning of this test. I guess we want to test that specifying all the options are present in the resulting strings. Do we want to make sure that empty or 0 values are ok? Isn't this the test that should be caught by the config validation?

Reply:
I want to ensure that whatever the resource builder in the driver is doing, we can assert that the result is correct and parsable as colonseparated list of key=value pairs.

The "0" part of the test is confusing, thus the comment. I am just using it to be able to send `None` arguments to the driver.